src/main/java/com/exactpro/remotehand/web/WebAction.java

```diff
@@ -22,118 +22,122 @@
 	}
-	
-	public static int getIntegerParam(Map<String, String> params, String paramName) throws ScriptExecuteException
-	{
-		try
-		{
+
+	public static int getIntegerParam(Map<String, String> params, String paramName) throws ScriptExecuteException {
+		try {
 			return Integer.parseInt(params.get(paramName));
-		}
-		catch (NumberFormatException e)
-		{
+		} catch (NumberFormatException e) {
 			throw new ScriptExecuteException(String.format("Error while parsing parameter '%s' = '%s' as number",
 					paramName, params.get(paramName)), e);
 		}
 	}
 
-	public static int getIntegerParam(Map<String, String> params, String paramName, int defaultValue)
-	{
-		try
-		{
+	public static int getIntegerParam(Map<String, String> params, String paramName, int defaultValue) {
+		try {
 			return Integer.parseInt(params.get(paramName));
-		}
-		catch (NumberFormatException e)
-		{
+		} catch (NumberFormatException e) {
 			return defaultValue;
 		}
 	}
-	
-	public abstract boolean isNeedLocator();
-	public abstract boolean isCanWait();
-	public abstract String run(WebDriver webDriver, By webLocator, Map<String, String> params) throws ScriptExecuteException;
-	protected abstract Logger getLogger();
 
+	public boolean isLocatorNeeded() {
+		return locatorNeeded;
+	}
 
-	public boolean isCanSwitchPage()
-	{
-		return false;
+	public boolean isCanWait() {
+		return canWait;
 	}
-	
-	public boolean isElementMandatory()
-	{
-		return !params.containsKey(PARAM_NOTFOUNDFAIL) || RhUtils.YES.contains(params.get(PARAM_NOTFOUNDFAIL));
+
+	public abstract String run(WebDriver webDriver, By webLocator, Map<String, String> params) throws ScriptExecuteException;
+
+	public boolean isCanSwitchPage() {
+		return canSwitchPage;
+	}
+
+	public boolean isElementMandatory() {
+		return !params.containsKey(PARAM_NOT_FOUND_FAIL) || RhUtils.YES.contains(params.get(PARAM_NOT_FOUND_FAIL));
 	}
-	
-	protected boolean waitForElement(WebDriver webDriver, int seconds, By webLocator) throws ScriptExecuteException
-	{
+
+	protected boolean waitForElement(WebDriver webDriver, int seconds, By webLocator) throws ScriptExecuteException {
 		return waitForElement(webDriver, seconds, webLocator, isElementMandatory());
 	}
-	
-	protected boolean waitForElement(WebDriver webDriver, int seconds, By webLocator, boolean isElementMandatory)
-			throws ScriptExecuteException
-	{
-		try
-		{
+
+	protected boolean waitForElement(
+			WebDriver webDriver,
+			int seconds,
+			By webLocator,
+			boolean isElementMandatory
+	) throws ScriptExecuteException {
+		try {
 			new WebDriverWait(webDriver, seconds).until((ExpectedCondition<Boolean>) webDriver1 ->
-					webDriver1 != null && webDriver1.findElements(webLocator).size() > 0);
-			logInfo("Appeared locator: '%s'", webLocator);
-		}
-		catch (TimeoutException ex)
-		{
+					webDriver1 != null && !webDriver1.findElements(webLocator).isEmpty()
+			);
+			if (logger.isInfoEnabled()) {
+				logger.info("Appeared locator: '{}'", webLocator);
+			}
```

Comment:
```suggestion
			logger.info("Appeared locator: '{}'", webLocator);
```
We can avoid isInfoEnabled check, because webLocator is passed as object